source/extensions/filters/network/redis_proxy/command_splitter_impl.cc

```diff
@@ -153,18 +110,17 @@ SplitRequestPtr SimpleRequest::create(Router& router,
 
   const auto route = router.upstreamPool(incoming_request->asArray()[1].asString());
   if (route) {
-    request_ptr->conn_pool_ = route->upstream();
+    Common::Redis::RespValueSharedPtr base_request = std::move(incoming_request);
     request_ptr->handle_ =
-        makeRequest(route, incoming_request->asArray()[0].asString(),
-                    incoming_request->asArray()[1].asString(), *incoming_request, *request_ptr);
+        makeSingleServerRequest(route, base_request->asArray()[0].asString(),
+                                base_request->asArray()[1].asString(), base_request, *request_ptr);
```

Comment:
possible simplication idea- do `base_request->asArray()[X].asString()` inside `makeSingleServerRequest`?

Reply:
I guess that doesn't work though with the way that we have commands and keys for `EVAL` though...